### PR TITLE
Added `regexp/no-trivially-nested-assertion` rule

### PR DESCRIPTION
**Pull request by RunDevelopment**
This is the `clean-regex/no-trivially-nested-lookaround` rule but with a different name and [this bug](https://github.com/ota-meshi/eslint-plugin-regexp/issues/91#issuecomment-817773291) fixed.

I renamed it because the "trivially nested assertion" is the inner assertion (e.g. in `/(?=(?!a))/`, the trivially nested assertion is `(?!a)`). Since the inner assertion can be any assertion and not just a lookaround, I renamed the rule.

### CODE REVIEW

File: lib/rules/no-trivially-nested-assertion.ts
Comment (ota-meshi) [inline lib/rules/no-trivially-nested-assertion.ts:65]:
```suggestion
            // TODO Switch to recommended in the major version.
            // recommended: true,
            recommended: false,
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR! Almost LGTM.

Is it better to change the `new` script?

**Review by ota-meshi — APPROVED** [review]
Thank you for changing it!

**Comment by RunDevelopment** [comment]
@ota-meshi Do you have a merge policy? I just merge past PRs as soon as they were approved and mergeable to move quickly. Should I not have done that?

**Comment by ota-meshi** [comment]
I was wondering when to merge 😅.
You and I are the two maintainers of this plugin, so do you think it would be good to merge them as soon as we `Approve`?
If so, I will merge as soon as I think your PR is good.

**Comment by RunDevelopment** [comment]
> You and I are the two maintainers of this plugin

No, I'm currently only a collaborator. This project isn't in an organization, so there can't be any maintainers from my understanding.

> do you think it would be good to merge them as soon as we Approve?

I think it is. No need to keep PRs open if you think they are good to go.

If the author of a PR thought that the PR still needs some changes, then the author should either say that or convert the PR into a draft.

---

I'll merge this now since you approved :)

**Comment by ota-meshi** [comment]
> No, I'm currently only a collaborator. This project isn't in an organization, so there can't be any maintainers from my understanding.

I didn't know much about it. Do you think you should move to an organization for this project?  
(I'm not familiar with how to create an GitHub organization yet.)

**Comment by RunDevelopment** [comment]
> Do you think this project should move to an organization for this project?

AFAIK, the main reasons people make organizations are:

1. Organization multiple projects.
1. Organizing teams/people. (This is what makes [maintainers](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/giving-team-maintainer-permissions-to-an-organization-member).)
1. Moneatziation/Sponsioring.

We don't have multiple projects or sponsoring right now, so that's not an issue. If more people joined or you wanted to someday stop working on the project, then we can make an organization. Until then, I don't think we need an organization.

The only difference between maintainers and collaborators is that collaborators can't access the settings tab. So it's not really a hindrance, right now. 

That being said, I very much appreciate your offer.

**Comment by ota-meshi** [comment]
Thank you for sharing your thoughts.

> Until then, I don't think we need an organization.

I understand. If we think we need something, let us talk again.

**Pull request merged**
